src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataRpcExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Text.Json;
+using Google.Protobuf.Collections;
+using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+
+namespace Microsoft.Azure.Functions.Worker.Grpc.FunctionMetadata
+{
+    internal static class FunctionMetadataRpcExtensions
+    {
+        internal static MapField<string, BindingInfo> GetBindingInfoList(this IFunctionMetadata funcMetadata)
+        {
+            if (funcMetadata is RpcFunctionMetadata rpcFuncMetadata)
+            {
+                return rpcFuncMetadata.Bindings;
+            }
+
+            MapField<string, BindingInfo> bindings = new MapField<string, BindingInfo>();
+            var rawBindings = funcMetadata.RawBindings;
+
+            if(rawBindings.Count == 0)
+            {
+                throw new FormatException("At least one binding must be declared in a Function.");
```

Comment:
`FormatException` - is this our exception type? Or is it the one from the BCL? If it is BCL, I believe that exception is mostly for string formatting. A more appropriate exception type might be `InvalidOperationException`.

Reply:
I think reporting [diagnostic errors](https://github.com/Azure/azure-functions-dotnet-worker/blob/45555ee1ca5d44fef1fe62b32e8f202d9e552c38/sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs#L154) is the approach and it's used in the other source-generator in this repo. Will work on refactoring to use this in the next iteration (after servicing 😄 )